src/shared/find-esp.c

```diff
@@ -359,13 +346,24 @@ static int verify_esp(
         /* Non-root user can only check the status, so if an error occurred in the following, it does not cause any
          * issues. Let's also, silence the error messages. */
 
+        r = chaseat(rfd, path, CHASE_AT_RESOLVE_IN_ROOT|CHASE_PARENT, &p, &pfd);
+        if (r < 0)
+                return log_full_errno((searching && r == -ENOENT) ||
+                                      (unprivileged_mode && ERRNO_IS_PRIVILEGE(r)) ? LOG_DEBUG : LOG_ERR,
+                                      r, "Failed to open parent directory of \"%s\": %m", path);
+
+        r = path_extract_filename(p, &f);
```

Comment:
Then, now this is only for `xstatfsat()` below, but it can be call with `rfd` and `p`, i.e. `xstatfsat(rfd, p, &sfs)`. Hence, this can be dropped. At least, this can be moved to `if (!relax_checks)` block.

Reply:
It can't, we might follow symlinks outside of the root directory if we did that. We need to chase first.

Reply:
I can move it to inside the if block though.